main/lsp/LSPQuery.h

```diff
@@ -0,0 +1,21 @@
+#ifndef SORBET_LSP_QUERY_H
+#define SORBET_LSP_QUERY_H
+
+#include "main/lsp/LSPTypechecker.h"
+
+namespace sorbet::realmain::lsp {
+
+std::vector<std::unique_ptr<core::lsp::QueryResponse>>
+filterAndDedup(const core::GlobalState &gs,
+               const std::vector<std::unique_ptr<core::lsp::QueryResponse>> &queryResponses);
+
+LSPQueryResult queryByLoc(const LSPConfiguration &config, LSPTypecheckerInterface &typechecker, std::string_view uri,
+                          const Position &pos, LSPMethod forMethod, bool errorIfFileIsUntyped = true);
+LSPQueryResult queryBySymbolInFiles(const LSPConfiguration &config, LSPTypecheckerInterface &typechecker,
+                                    core::SymbolRef symbol, std::vector<core::FileRef> frefs);
+LSPQueryResult queryBySymbol(const LSPConfiguration &config, LSPTypecheckerInterface &typechecker,
+                             core::SymbolRef symbol);
+
+} // namespace sorbet::realmain::lsp
+
+#endif
```

Comment:
Given that this file is called `LSPQuery`, what do you think about moving these methods to either be public static methods (or methods inside a namspace) on a namespace called `LSPQuery`?

```cpp
LSPQuery::byLoc(...)
LSPQuery::bySymbol(...)
```

Reply:
Thanks, that's a great suggestion! I was thinking about grouping those functions into the object, but didn't see any clear value of doing that. But with a rename you proposed it makes much more sense semantically